utils/transformedPayload.js

```diff
@@ -0,0 +1,24 @@
+const transformPayload = (payload, referenceData) => {
```

Comment:
As your application grows this function might be used in other parts of your code. Think of it like a utility function and can you add unit tests to test this function for positive and negative scenarios.